doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1023 @@
+- **sai_vlan_attr_t** - It will be enhanced to have one new attribute
+  - SAI_VLAN_ATTR_NEIGH_SUPP_MODE - By default this value will be FALSE. Reason for this new attribute is due to recent changes in ARP behaviour. The default arp behavior in copp file is changed from trap to copy(as part of commit -5e4b71d4). But for ARP Suppression, we need Pkt to get trapped instead of copy to avoid flooding of packets over vxlan tunnel.So a differentiated tcam rule for these arp suppressed vlans will redirect the ARP pkts to cpu instead of copy.
+
+The changes are subject to approval by the SAI community.
+
+## 4.2 DB Changes
+
+At a high level below are some of the interactions between relevant components and the DB involved for EVPN L2 VXLAN support in SONiC architecture.
+
+![EVPN L2 VXLAN Arch](images/arch.PNG "Figure : Arch")
+__Figure 4: EVPN L2 VXLAN DBs__
+
+### 4.2.1 CONFIG_DB changes
+
+**EVPN_NVO_TABLE**
+
+Producer:  config manager 
+
+Consumer: VxlanMgr
+
+Description: This is a singleton configuration which holds the VTEP source IP used for BGP-EVPN discovered tunnels. 
+
+Schema:
+
+```
+;New table
+;holds the VTEP source IP used for BGP-EVPN discovered tunnels
+
+key = EVPN_NVO_TABLE:nvo_name
+                          ; nvo or VTEP name as a string
+; field = value
+source_vtep = tunnel_name ; refers to the tunnel object created with SIP only.
+```
+
+**NEIGH_SUPPRESS_CFG_VLAN_TABLE**
+
+Producer:  config manager
+
+Consumer: VlanMgr
+
+Description: New table that stores neighbor suppression per VLAN configuration.
+
+Schema:
+
+```
+;New table
+;Stores Neigh Suppress configuration per VLAN
+
+key             = NEIGH_SUPPRESS_VLAN:"Vlan"vlanid       ; VLAN with prefix "NEIGH_SUPPRESS"
+suppress_neigh  = "ON" / "OFF" ; Default "OFF"
+
+```
+
+**VRF_TABLE**
+
+Producer:  config manager
+
+Consumer: VrfMgr
+
+Description: Updated existing table to store VNI associated with VRF.
+
+Schema:
+
+```
+;Existing table
+;defines virtual routing forward table. Updated to stores VNI associated with VRF
+;
+;Status: stable
+
+key = VRF_TABLE:VRF_NAME ;
+fallback = "true"/"false"
+vni = 1*8DIGIT ; VNI associated with VRF
+
+```
+
+### 4.2.2 APP_DB Changes
+
+**EVPN_NVO_TABLE**
+
+Producer:  VxlanMgr 
+
+Consumer: VxlanOrch
+
+Description: This is a singleton configuration which holds the VTEP source IP used for BGP-EVPN discovered tunnels. 
+
+Schema:
+
+```
+; New table
+; holds the VTEP source IP used for BGP-EVPN discovered tunnels
+
+key = EVPN_NVO_TABLE:nvo_name
+                          ; nvo or VTEP name as a string
+; field = value
+source_vtep = tunnel_name ; refers to the VXLAN_TUNNEL object created with SIP only.
+```
+
+**EVPN_REMOTE_VNI** 
+
+Producer: fdbsyncd
+
+Consumer: VxlanOrch
+
+Description: This specifies the remote VNI to be extended over a tunnel. This is populated by fdbsyncd on receipt of a IMET route from the remote side. 
+
+Schema:
+
+```
+; New table
+; specifies the VLAN to be extended over a tunnel
```

Comment:
This table is confusing. For a given local VLAN the VNI is mapped. If an IMET route is received with a matching VNI, we can build a mapping of VLAN,VNI to list of remote VTEPs.  
But the current schema allows a combination of <V1,RIP1>, <V2,RIP1> map to same VNI.

Reply:
This is to handle the case of downstream assigned VNI in DCI cases in future. The VNI in the table specifies the VNI carried in the IMET route. This VNI will be used during encap. The remote will advertise unique VNI values per VLAN. In the DCI case the remote VNI will be different from the local VNI for a VLAN whereas in a global VNI case it will be the same.